File: app/src/main/java/com/devdream/cookall/main/DummyListContent.java
package com.devdream.cookall.main;

import com.devdream.cookall.core.dto.RecipeDTO;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class DummyListContent {

    public static final List<RecipeDTO> ITEMS = new ArrayList<RecipeDTO>();

    public static final Map<Long, RecipeDTO> ITEM_MAP = new HashMap<Long, RecipeDTO>();

    private static final int COUNT = 10;

    static {
        // Add some sample items.
        for (int i = 1; i <= COUNT; i++) {
            addItem(createDummyItem(i));
        }
    }

    private static void addItem(RecipeDTO recipe) {
        ITEMS.add(recipe);
        ITEM_MAP.put(recipe.id, recipe);
    }

    private static RecipeDTO createDummyItem(int position) {
        RecipeDTO recipe = new RecipeDTO();
        recipe.id = position;
        recipe.title = "Details about Item: " + position;
        return recipe;
    }

}

File: app/src/main/java/com/devdream/cookall/main/DummyListContent.kt
package com.devdream.cookall.main

import com.devdream.cookall.core.dto.RecipeDTO

import java.util.ArrayList
import java.util.HashMap

object DummyListContent {

    val ITEMS: MutableList<RecipeDTO> = ArrayList()

    val ITEM_MAP: MutableMap<Long, RecipeDTO> = HashMap()

    private val COUNT = 10

    init {
        // Add some sample items.
        for (i in 1..COUNT) {
            addItem(createDummyItem(i))
        }
    }

    private fun addItem(recipe: RecipeDTO) {
        ITEMS.add(recipe)
        ITEM_MAP[recipe.id] = recipe
    }

    private fun createDummyItem(position: Int): RecipeDTO {
        val recipe = RecipeDTO()
        recipe.id = position.toLong()
        recipe.title = "Details about Item: $position"
        return recipe
    }

}
